lib/index.js

```diff
@@ -459,6 +463,18 @@ Socket.prototype.rmchan     = function() {
   };
 }
 
+Socket.prototype.dontwait   = function (n) {
+  if ( arguments.length ) {
+    if (typeof n !== 'number' || n > 1 || n < 0) {
+      throw new TypeError('dontwait option must be either 1 or 0')
+    }
+    this.nndontwait = n;
+    return true;
+  } else {
+    return this.nndontwait;
+  }
```

Comment:
this function now can return a number or bool.  Not that someone's going to call it a lot, but we should stick with one return type (bool).